backend/src/main/java/ro/utcn/sd/mdantonio/StackUnderflow/controller/UserController.java

```diff
@@ -0,0 +1,50 @@
+package ro.utcn.sd.mdantonio.StackUnderflow.controller;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.*;
+import ro.utcn.sd.mdantonio.StackUnderflow.dto.UserDTO;
+import ro.utcn.sd.mdantonio.StackUnderflow.service.UnderflowUserManagementService;
+import ro.utcn.sd.mdantonio.StackUnderflow.service.UserSessionManagementService;
+
+import java.util.List;
+
+import static ro.utcn.sd.mdantonio.StackUnderflow.entities.StackUnderflowConstants.USER;
+
+@Slf4j
+@RestController
+@RequiredArgsConstructor
+public class UserController {
+    private final UnderflowUserManagementService underflowUserManagementService;
+    private final UserSessionManagementService userSessionManagementService;
+
+    @PostMapping("/register")
+    public UserDTO registerUser(@RequestBody UserDTO dto) {
+        return underflowUserManagementService.addUnderflowUser(dto.getUsername(), dto.getPassword(), dto.getEmail(),
+                false, USER);
+    }
+
+    @PostMapping("/login")
+    public UserDTO login(@RequestBody UserDTO dto) {
```

Comment:
You already use HTTP Basic (in the security configuration), so you don't need to check the password once again in your own custom login controller.